src/chunk.c

```diff
@@ -2659,9 +2660,22 @@ chunk_tuple_delete(TupleInfo *ti, DropBehavior behavior, bool preserve_chunk_cat
 					ts_dimension_slice_scan_by_id_and_lock(cc->fd.dimension_slice_id,
 														   &tuplock,
 														   CurrentMemoryContext);
-				if (ts_chunk_constraint_scan_by_dimension_slice_id(slice->fd.id,
-																   NULL,
-																   CurrentMemoryContext) == 0)
+				/* If the slice is not found in the scan above, the table is
+				 * broken so we do not delete the slice. We proceed with
+				 * anyway since users need to be able to drop broken tables or
+				 * remove broken chunks. */
```

Comment:
```suggestion
				 * broken so we do not delete the slice. We proceed 
				 * anyway since users need to be able to drop broken tables or
				 * remove broken chunks. */
```